src/main/java/com/spotify/reaper/storage/CassandraStorage.java

```diff
@@ -446,8 +487,10 @@ public int compare(RepairSegment seg1, RepairSegment seg2) {
               (segmentIsWithinRange(seg, range.get()))
               ) || !range.isPresent()) // Token range condition
           ){
-        segment = seg;
-        break;
+        if(takeLeadOnSegment(seg.getId())) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)